### DISCUSSION TITLE
How to automaticlly rerun the zombie tasks

### DISCUSSION THREAD

**Discussion by linhgao** [apache/airflow#29463] [Q&A]
### Description

My airflow cluster is deployed on the k8s cluster, when the worker container is restarted, the long-run jobs will failed. 
And now I only rerun manually, if I don't rerun in time, it will affect the normal operation of workflow.

So, I want to achieve the feature that the airflow scheduler can automaticlly rerun the zombie tasks.

### Use case/motivation

When the scheduler find the zombie tasks, we can choose by the airflow configuration that mark failed these tasks or automaticlly add these task into queue.

### Related issues

no

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by hussein-awala** [comment]
Based on the [doc](https://airflow.apache.org/docs/apache-airflow/stable/core-concepts/tasks.html#zombie-undead-tasks):
> Zombie tasks are tasks that are supposed to be running but suddenly died (e.g. their process was killed, or the machine died). Airflow will find these periodically, clean them up, and either fail or retry the task depending on its settings.

the zombie tasks are passed to `up_for_retry` if they didn't reach the `retries` which is `core.default_task_retries` or 0 by default, so if you have `retries` > 0 and the zombie tasks are passed directly to `failed`, it could be a bug.

Can you please check what configuration you have?
